### PR TITLE
[IMPROVEMENT] Update runCI from NFS share

### PR DESCRIPTION
**Pull request by TheClashster**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

Have added a variable for it as well as pulling the specific file from the origin, I hope this is the correct method.

### CODE REVIEW

File: install/ci-vm/ci-linux/ci/bootstrap [hunk 1]
```diff
@@ -20,6 +20,7 @@ if [ "${maintenance}" == "True" ]; then
     exit;
 fi
 
-wget "${runCIURL}"
+# Copy runCI to local folder
+cp ${runCIFile} $DIR/runCI
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/bootstrap]: If $DIR contains spaces this won't work as intended.

File: mod_deploy/controllers.py [hunk 2]
```diff
@@ -135,6 +135,11 @@ def deploy():
         with open('build_commit.py', 'w') as f:
             f.write(build_commit)
 
+        # Update runCI
+        runCI = origin.pull('install/ci-vm/ci-linux/ci/runCI').read()
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: Why are you doing this?

File: mod_deploy/controllers.py [hunk 3]
```diff
@@ -135,6 +135,11 @@ def deploy():
...
+        runCI_path = os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'vm_data', 'runCIFile')
+        runCI.getroottree().write(os.path.join(runCI_path, 'runCI'), encoding='utf-8', pretty_print=True)
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: `getroottree()`? That seems to be a bit of careless copying...

File: mod_deploy/controllers.py [hunk 4]
```diff
@@ -135,6 +135,11 @@ def deploy():
         with open('build_commit.py', 'w') as f:
             f.write(build_commit)
 
+        # Update runCI
+        runCI = origin.pull('install/ci-vm/ci-linux/ci/runCI').read()
+        runCI_path = os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'vm_data', 'runCIFile')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: I'm missing the linux name in here.

File: mod_deploy/controllers.py [hunk 5]
```diff
@@ -135,6 +136,13 @@ def deploy():
         with open('build_commit.py', 'w') as f:
             f.write(build_commit)
 
+        # Update runCI
+        runCIrepo = config.get('install/ci-vm/ci-linux/ci/runCI', '')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: What is this for? This is not a defined variable, and won't be either.

File: mod_deploy/controllers.py [hunk 6]
```diff
@@ -135,6 +136,13 @@ def deploy():
...
+        kvm_name = config.get('KVM_LINUX_NAME', '')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: You can inline this.

File: mod_deploy/controllers.py [hunk 7]
```diff
@@ -135,6 +136,13 @@ def deploy():
...
+        runCI_path = os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'vm_data', kvm_name, 'runCI')
+        runCI_nfs = os.path.join(runCI_path, 'runCI')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: Why do you have the runCI defined twice? Also, why not join that all together?

File: mod_deploy/controllers.py (hunk 5 above)
Comment (canihavesomecoffee) [reply mod_deploy/controllers.py]: If this is supposed to be the path, it's not ok.

File: install/ci-vm/ci-linux/ci/bootstrap [hunk 8]
```diff
@@ -20,6 +20,7 @@ if [ "${maintenance}" == "True" ]; then
     exit;
 fi
 
-wget "${runCIURL}"
+# Copy runCI to local folder
+cp ${runCIFile} $DIR/'runCI'
```
Comment (canihavesomecoffee) [inline install/ci-vm/ci-linux/ci/bootstrap]: Putting quotes there won't help.

File: mod_deploy/controllers.py [hunk 9]
```diff
@@ -135,6 +136,11 @@ def deploy():
         with open('build_commit.py', 'w') as f:
             f.write(build_commit)
 
+        # Update runCI
+        runCIrepo = os.path.join(repo,'install/ci-vm/ci-linux/ci/runCI')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: No, `repo` is not a path string.

Comment (canihavesomecoffee) [reply mod_deploy/controllers.py]: Also, please split up the other string. We're using path.join, so use it for all parts.

File: mod_deploy/controllers.py [hunk 10]
```diff
@@ -135,6 +136,11 @@ def deploy():
...
+        runCI_path = os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'vm_data', config.get('KVM_LINUX_NAME', ''), 'runCI')
+        copyfile(runCIrepo, runCI_nfs)
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: `runCI_nfs` is not defined.

File: mod_deploy/controllers.py [hunk 11]
```diff
@@ -135,6 +136,12 @@ def deploy():
         with open('build_commit.py', 'w') as f:
             f.write(build_commit)
 
+        # Update runCI
+        runCI_repo = os.path.join(app.config['INSTALL_FOLDER'], 'install', 'ci-vm', 'ci-linux', 'ci', 'runCI')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: please change the naming of the variables.

File: mod_deploy/controllers.py [hunk 12]
```diff
@@ -135,6 +136,12 @@ def deploy():
...
+        runCI_nfs = os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'vm_data', config.get(
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: config is undefined in this context.

Comment (canihavesomecoffee) [reply mod_deploy/controllers.py]: so is os.

File: mod_deploy/controllers.py [hunk 13]
```diff
@@ -137,10 +138,10 @@ def deploy():
             f.write(build_commit)
 
         # Update runCI
-        runCI_repo = os.path.join(app.config['INSTALL_FOLDER'], 'install', 'ci-vm', 'ci-linux', 'ci', 'runCI')
-        runCI_nfs = os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'vm_data', config.get(
```
Comment (saurabhshri) [inline mod_deploy/controllers.py]: Why is it change from `app.config` --> `config`?

File: mod_deploy/controllers.py [hunk 14]
```diff
@@ -8,6 +8,7 @@
 import hmac
 import json
 import subprocess
+import os
```
Comment (saurabhshri) [inline mod_deploy/controllers.py]: Is it needed outside the `deploy()` function?

File: mod_deploy/controllers.py (hunk 13 above)
Comment (TheClashster) [reply mod_deploy/controllers.py]: app.config is not defined in current context I guess.

File: mod_deploy/controllers.py (hunk 14 above)
Comment (TheClashster) [reply mod_deploy/controllers.py]: Nope.

File: mod_deploy/controllers.py [hunk 15]
```diff
@@ -135,6 +137,12 @@ def deploy():
         with open('build_commit.py', 'w') as f:
             f.write(build_commit)
 
+        # Update runCI
+        run_ci_repo = os.path.join(config['INSTALL_FOLDER'], 'install', 'ci-vm', 'ci-linux', 'ci', 'runCI')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: config is undefined.

File: mod_deploy/controllers.py [hunk 16]
```diff
@@ -135,6 +137,12 @@ def deploy():
         with open('build_commit.py', 'w') as f:
             f.write(build_commit)
 
+        # Update runCI
+        run_ci_repo = path.join(app.config['INSTALL_FOLDER'], 'install', 'ci-vm', 'ci-linux', 'ci', 'runCI')
+        run_ci_nfs = path.join(app.config['SAMPLE_REPOSITORY'], 'vm_data', app.config[
+            'KVM_LINUX_NAME'], 'runCI')
```
Comment (canihavesomecoffee) [inline mod_deploy/controllers.py]: Why is it broken of like this? Why not just the 'runCI' part?

Comment (TheClashster) [reply mod_deploy/controllers.py]: Fixed 😃

**Pull request merged**